agent/util-scripts/pbench-register-tool-trigger

```diff
@@ -19,16 +20,23 @@ stop_trigger=""
 
 function usage() {
 	printf "usage:\n"
-	printf "$script_name --group=<tool group> --start-trigger=<string to match> --stop-trigger=<string to match>"
+	printf "$script_name [--group=<tool group>] --start-trigger=<string to match> --stop-trigger=<string to match>\n"
 }
 
+case $# in
+    0)
+        usage >&2
+        exit 1
+        ;;
+esac
+
 # Process options and arguments
 opts=$(getopt -q -o h --longoptions "help,group:,start-trigger:,stop-trigger:" -n "getopt.sh" -- "$@");
 if [ $? -ne 0 ]; then
-	printf "\n"
-	printf "%s\n" $*
-	printf "$script_name: you specified an invalid option\n\n"
-	usage
+	printf "\n" >&2
+	printf "%s\n" $* >&2
```

Comment:
This is really an existing bug, but since you are making changes here ... the `$*` needs to be quoted to make sure it is expanded to one argument.

Here is a demonstration of the bad behavior:

```
$ cat /tmp/testit
#!/bin/bash

printf "%s\n" $*
printf "%s\n" "${*}"
$ /tmp/testit a b c e
a
b
c
e
a b c e
$
```

Perhaps we can collapse this to one `printf`:

```
printf "\n%s\n%s: you specified an invalid option\n\n" "${*}" "${script_name}" >&2
```

Reply:
I combined them into one `printf` with proper quoting (I hope): I should have done that in the first place instead of adding `>&2` umpteen times.